src/components/TodoList.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from './TodosContext';
+
+export const TodoList: React.FC = () => {
+  const { filteredTodos, todos, setTodos } = useContext(TodosContext);
+
+  const toggleTodos = () => {
+    let newTodos;
+
+    if (todos.some((item) => !item.completed)) {
+      newTodos = todos.map((item) => {
+        if (!item.completed) {
+          return {
+            ...item,
+            completed: true,
+          };
+        }
+
+        return item;
+      });
+    } else {
+      newTodos = todos.map((item) => {
+        return {
+          ...item,
+          completed: false,
+        };
+      });
+    }
+
+    setTodos(newTodos);
+  };
+
+  return (
+    <section className="main">
+      <input
+        type="checkbox"
+        id="toggle-all"
+        className="toggle-all"
+        data-cy="toggleAll"
+        onClick={() => toggleTodos()}
+        defaultChecked={todos.every((item) => item.completed)}
```

Comment:
please don't call a function from the template because it will be executed every time when the state change, so make sense to move it to the variable

please check other places